src/status_im2/contexts/chat/camera/view.cljs

```diff
@@ -0,0 +1,94 @@
+(ns status-im2.contexts.chat.camera.view
+  (:require
+    [quo2.core :as quo]
+    [quo2.foundations.colors :as colors]
+    [react-native.camera-kit :as camera-kit]
+    [react-native.core :as rn]
+    [react-native.fast-image :as fast-image]
+    [react-native.reanimated :as reanimated]
+    [react-native.safe-area :as safe-area]
+    [reagent.core :as reagent]
+    [utils.i18n :as i18n]
+    [status-im2.contexts.chat.camera.style :as style]
+    [utils.re-frame :as rf]))
+
+(defn- f-zoom-button
+  [{:keys [value current-zoom]}]
+  (let [selected (= @current-zoom value)
+        size     (reanimated/use-shared-value (if selected 37 25))]
+    (rn/use-effect #(reanimated/animate size (if selected 37 25)) [@current-zoom])
+    [rn/touchable-opacity
+     {:on-press #(reset! current-zoom value)
+      :style    style/zoom-button-container}
+     [reanimated/view {:style (style/zoom-button size)}
+      [quo/text
+       {:size   (if selected :paragraph-2 :label)
+        :weight :semi-bold
+        :style  {:color (if selected
+                          colors/system-yellow
+                          colors/white)}}
+       (str value (when selected "x"))]]]))
+
+(defn zoom-button
+  [args]
+  [:f> f-zoom-button args])
+
+(defn snap-button
+  [camera-ref uri]
+  [rn/view {:style style/outer-circle}
+   [rn/touchable-opacity
+    {:on-press (fn [] (camera-kit/capture @camera-ref #(reset! uri %)))
+     :style    style/inner-circle}]])
+
+(defn camera-screen
+  []
+  (let [camera-ref   (atom nil)
+        uri          (reagent/atom nil)
+        current-zoom (reagent/atom "1")]
+    (fn []
+      (let [window                 (rn/get-window)
+            {:keys [width height]} window
+            camera-window-height   (* width 1.33)
+            insets                 (safe-area/get-insets)
+            top                    (/ (- height camera-window-height (:bottom insets)) 2)]
+        [rn/view {:style style/screen-container}
+         [rn/view {:style style/flash-container}
+          [quo/icon :i/flash-camera
+           {:color colors/white
+            :size  24}]]
+         (if @uri
+           [fast-image/fast-image
+            {:style  (style/camera-window width camera-window-height top)
+             :source {:uri @uri}}]
+           [camera-kit/camera
+            {:ref   #(reset! camera-ref %)
+             :style (style/camera-window width camera-window-height top)}])
+         [rn/view {:style (style/zoom-container top insets)}
```

Comment:
I believe the zoom control needs to be hidden on the preview.

<img src="https://github.com/status-im/status-mobile/assets/19339952/8dae0d24-2a86-4e90-a628-da8f8de7e837" width="300" />

Reply:
@smohamedjavid Yes, and the flash button also. Completed in a second issue https://github.com/status-im/status-mobile/issues/16580